src/Lecture2.hs

```diff
@@ -208,9 +238,18 @@ The algorithm of merge sort is the following:
 [1,2,3]
 -}
 mergeSort :: [Int] -> [Int]
-mergeSort = error "TODO"
-
-
+mergeSort = go . inverseConcat
+  where
+    go :: [[Int]] -> [Int]
+    go []               = []
+    go [xs]             = xs
+    go (xs : ys : rest) = go (merge xs ys : rest)
```

Comment:
That's a clever and correct solution 😅 
Unfortunately, it's not optimal asymptotically. Your solution has complexity `O(n ^ 2)` while optimal merge sort has `O(n log n)`. But the standard `sort` function in Haskell implements an idea similar to yours and improves it a bit to achieve even better performance 🐎